libdnf/base/base.cpp

```diff
@@ -190,6 +190,12 @@ void Base::setup() {
 
     get_vars()->load(vars_installroot, config.get_varsdir_option().get_value());
 
+    if (!config.get_forcearch_option().get_value().empty()) {
+        config.get_ignorearch_option().set(Option::Priority::RUNTIME, true);
+        get_vars()->set("arch", config.get_forcearch_option().get_value(), libdnf::Vars::Priority::CONFIG);
+        get_vars()->set("basearch", config.get_forcearch_option().get_value(), libdnf::Vars::Priority::CONFIG);
+    }
```

Comment:
I am  not in favor of a redundancy because of this part of the code. It is possible to modify vars from multiple side including commandline. In the code above, you try to handle priority and this could be a problem. Lets use as a test command `dnf install acpi  --setvar=arch=x86_64 --forcearch=s390`. I believe that the final value would be `x86_64`, because it has command-line priority and cannot be overridden by the Config priority.